### PR TITLE
screen and widget things

### PR DESCRIPTION
**Pull request by eristhea**
work in progress!

### CODE REVIEW

File: mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping [hunk 1]
```diff
@@ -1,9 +1,48 @@
...
+	METHOD m_djhyilbe setPadding (II)Lnet/minecraft/unmapped/C_pjarmpvz;
+		ARG 1 horizontal
+		ARG 2 vertical
+	METHOD m_ewcvnloj setAlignment (FF)Lnet/minecraft/unmapped/C_pjarmpvz;
+		ARG 1 x
+		ARG 2 y
+	METHOD m_gouapxzs alignVerticallyTop ()Lnet/minecraft/unmapped/C_pjarmpvz;
+	METHOD m_hyvejfbf setPadding (I)Lnet/minecraft/unmapped/C_pjarmpvz;
+		ARG 1 padding
+	METHOD m_ietlmaqj alignVerticallyBottom ()Lnet/minecraft/unmapped/C_pjarmpvz;
```
Comment (EnnuiL) [inline mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping:14]: hm, `align<direction><axis>y` might be better

File: mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping [hunk 2]
```diff
@@ -1,9 +1,48 @@
...
+	METHOD m_yxhxelpo alignHorizontallyRight ()Lnet/minecraft/unmapped/C_pjarmpvz;
+	METHOD m_zygwkpxf setAlignmentX (F)Lnet/minecraft/unmapped/C_pjarmpvz;
+		ARG 1 x
+	CLASS C_efyrmufb Impl
 		FIELD f_hxiztipo leftPadding I
 		FIELD f_kvjbnbeh bottomPadding I
 		FIELD f_lgffdybo rightPadding I
 		FIELD f_uddwylwg alignmentX F
 		FIELD f_vvwvvykx topPadding I
 		FIELD f_wlrjffdq alignmentY F
```
Comment (EnnuiL) [inline mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping]: hm, I don't remember what I used on my original mappings; I think I used `horizontalAligment` and `verticalAlignment`...?

**Review by EnnuiL — COMMENTED** [review]
I won't look deep into this PR yet, but oh dear, do I wish you good luck

File: mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping (hunk 2 above)
Comment (EnnuiL) [reply mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping]: wait, `alignmentX` and `alignmentY` *are* my names

File: mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping (hunk 1 above)
Comment (eristhea) [reply mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping:14]: mm, I like `alignHorizontallyLeft` over `alignLeftHorizontally`

**Comment by eristhea** [comment]
note that this pr does not achieve 100% coverage of screens and widgets, there remain some unfinished screens, unmapped screens, and unmapped widgets
I intend to cover those later

**Comment by EnnuiL** [comment]
> note that this pr does not achieve 100% coverage of screens and widgets, there remain some unfinished screens, unmapped screens, and unmapped widgets I intend to cover those later

Ah, understandable! I hope to somehow help with these later

**Review by OroArmor — APPROVED** [review]

File: mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping [hunk 3]
```diff
@@ -1,9 +1,48 @@
...
+	METHOD m_vyorthoe setBottomPadding (I)Lnet/minecraft/unmapped/C_pjarmpvz;
+		ARG 1 padding
+	METHOD m_wtmggfev copy ()Lnet/minecraft/unmapped/C_pjarmpvz;
+	METHOD m_xkbxtfqc setVerticalAlignment (F)Lnet/minecraft/unmapped/C_pjarmpvz;
+		ARG 1 y
+	METHOD m_ygbfijsh alignHorizontallyLeft ()Lnet/minecraft/unmapped/C_pjarmpvz;
+	METHOD m_yxhxelpo alignHorizontallyRight ()Lnet/minecraft/unmapped/C_pjarmpvz;
+	METHOD m_zygwkpxf setHorizontalAlignment (F)Lnet/minecraft/unmapped/C_pjarmpvz;
+		ARG 1 x
+	CLASS C_efyrmufb Impl
```
Comment (EnnuiL) [inline mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping:40]: hm, this name feels weird, even though it makes sense

**Review by EnnuiL — APPROVED** [review]
This is what we've been needing for so long

File: mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping (hunk 3 above)
Comment (eristhea) [reply mappings/net/minecraft/client/gui/widget/container/LayoutSettings.mapping:40]: I *really* wanted to name is builder but couldn't

**Pull request merged**